src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {View, Pressable, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import ReportActionPropTypes from './ReportActionPropTypes';
+import styles from '../../../styles/styles';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+import {editReportComment, saveReportActionDraft} from '../../../libs/actions/Report';
+import {TouchableOpacity} from 'react-native-web';
+
+const propTypes = {
+    // All the data of the action
+    action: PropTypes.shape(ReportActionPropTypes).isRequired,
+
+    // Draft message
+    draftMessage: PropTypes.string.isRequired,
+
+    reportID: PropTypes.number.isRequired,
+};
+
+class ReportActionItemMessageEdit extends React.Component {
+    constructor(props) {
+        super(props);
+        this.updateDraft = this.updateDraft.bind(this);
+        this.deleteDraft = this.deleteDraft.bind(this);
+        this.debouncedSaveDraft = _.debounce(this.debouncedSaveDraft.bind(this), 1000, false);
+        this.publishDraft = this.publishDraft.bind(this);
+        this.triggerSaveOrCancel = this.triggerSaveOrCancel.bind(this);
+
+        this.state = {
+            draft: this.props.draftMessage,
+        };
+    }
+
+    /**
+     * Update the value of the draft in Onyx
+     *
+     * @param {String} newDraft
+     */
+    updateDraft(newDraft) {
+        const trimmedNewDraft = newDraft.trim();
+        this.setState({draft: trimmedNewDraft});
+        this.debouncedSaveDraft(trimmedNewDraft);
+    }
+
+    deleteDraft() {
+        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, '');
+    }
+
+    debouncedSaveDraft() {
+        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, this.state.draft);
+    }
+
+    publishDraft() {
+        editReportComment(this.props.reportID, this.props.action, this.state.draft);
+        this.deleteDraft();
+    }
+
+    triggerSaveOrCancel(e) {
+        if (e && e.key === 'Enter' && !e.shiftKey) {
+            e.preventDefault();
+            this.publishDraft();
+        } else if (e && e.key === 'Escape') {
+            e.preventDefault();
+            this.deleteDraft();
+        }
+    }
+
+    render() {
+        return (
+            <View style={[styles.chatItemMessage]}>
+                <TextInputFocusable
+                    multiline
+                    ref={el => this.textInput = el}
+                    onChangeText={this.updateDraft} // Debounced saveDraftComment
+                    onKeyPress={this.triggerSaveOrCancel}
+                    defaultValue={this.props.draftMessage}
+                    maxLines={16} // This is the same that slack has
+                    style={[styles.textInput, styles.flex0]}
+                />
+                <View style={[styles.flexRow, styles.mt1]}>
+                    <TouchableOpacity style={[styles.button, styles.mr2]}>
```

Comment:
As stated in [this thread](https://expensify.slack.com/archives/C03TQ48KC/p1618355196232200), I would prefer to see something like this:

```jsx
<Pressable
    onPress={this.deleteDraft}
    style={({hovered, pressed}) => [
        styles.button,
        styles.mr2,
        getButtonBackgroundColorStyle(getButtonState(hovered, pressed)),
    ]}
/>
```

Reply:
Also, I see you switched to `Pressable` but didn't use the style callback w/ `getButtonBackgroundColorStyle(getButtonState(hovered, pressed))`

Reply:
We might need @Expensify/design's help to think around 2 things:
1. The normal button hover color is the same as the hover color for the comment:
1. There is a [`greenHover` color](https://github.com/Expensify/Expensify.cash/blob/5ec3b3f73b00f648e6d4315f2f519d2843a978d5/src/styles/themes/default.js#L26), however, what should we use for `greenPressed`?
![Kapture 2021-05-06 at 10 05 45](https://user-images.githubusercontent.com/4741899/117338766-cd9bf200-ae53-11eb-8a91-f8c8ce92a42c.gif)

Or perhaps better yet, do we _need_ button styles for hover for the edit comment buttons?